app/celery/tasks.py

```diff
@@ -1000,3 +1011,37 @@ def get_recipient_csv(job: Job, template: Template) -> RecipientCSV:
         placeholders=template.placeholders,
         max_rows=get_csv_max_rows(job.service_id),
     )
+
+
+def _acknowledge_notification(notification_type, template, receipt):
+    """
+    Acknowledge the notification has been saved to the DB and sent to the service.
+
+    Args:
+    notification_type: str
+        Type of notification being sent; either SMS_TYPE or EMAIL_TYPE
+    template: model.Template
+        Template used to send notification
+
+    Returns: None
+    """
+    if notification_type == SMS_TYPE:
+        if Config.FF_PRIORITY_LANES:
+            if template.process_type == PRIORITY:
+                sms_priority.acknowledge(receipt)
+            elif template.process_type == NORMAL:
+                sms_normal.acknowledge(receipt)
+            elif template.process_type == BULK:
+                sms_bulk.acknowledge(receipt)
+        else:
+            sms_queue.acknowledge(receipt)
+    elif notification_type == EMAIL_TYPE:
+        if Config.FF_PRIORITY_LANES:
+            if template.process_type == PRIORITY:
+                email_priority.acknowledge(receipt)
+            elif template.process_type == NORMAL:
+                email_normal.acknowledge(receipt)
+            elif template.process_type == BULK:
+                email_bulk.acknowledge(receipt)
+        else:
+            email_queue.acknowledge(receipt)
```

Comment:
This feels like many conditions, it might be simplified with a dictionary of priority as key and functions as value. That would emulate proper function dispatch... which then someone might say just use an OO pattern, but I don't see which objects we could reuse around for that.